clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1377,7 +1381,8 @@ void CXXNameMangler::mangleUnqualifiedName(const NamedDecl *ND,
     // <lambda-sig> ::= <template-param-decl>* <parameter-type>+
     //     # Parameter types or 'v' for 'void'.
     if (const CXXRecordDecl *Record = dyn_cast<CXXRecordDecl>(TD)) {
-      if (Record->isLambda() && Record->getLambdaManglingNumber()) {
+      if (Record->isLambda() && (Record->getLambdaManglingNumber() ||
+                                 Context.isUniqueNameMangler())) {
```

Comment:
Could be more obvious to read with `||` on this line instead.

Reply:
That is the output of Clang-Format.  For better or worse.

Reply:
I see... And it does not use BreakBeforeBinaryOperators(BOS_NonAssignment)